tajo-client/src/main/java/org/apache/tajo/client/QueryClient.java

```diff
@@ -106,7 +107,7 @@
 
   GetQueryResultResponse getResultResponse(QueryId queryId) throws TajoException;
 
-  TajoMemoryResultSet fetchNextQueryResult(final QueryId queryId, final int fetchRowNum) throws TajoException;
+  Future<TajoMemoryResultSet> asyncFetchNextQueryResult(final QueryId queryId, final int fetchRowNum);
```

Comment:
How about renaming it into ``fetchNextQueryResultAsync()``?

It would be more consistent for a method name, starting a verb. So, the method's full meaning will be 'fetch the next query result asynchronously'.